### PR TITLE
implement redux for table rendering

### PR DESCRIPTION
**Pull request by kgaikwad**
Fixes [RHIROS-19](https://issues.redhat.com/browse/RHIROS-19)

+ adding @karelhala into loop

### CODE REVIEW

File: src/Components/RosTable/redux/systemsTableReducer.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const initialState = {
+    loading: false,
+    systemsData: [],
+    systemError: {}
+};
+
+const systemsTableReducer = () => {
+    const tableReducer = (state = initialState, action) => {
+        switch (action.type) {
```
Comment (karelhala) [inline src/Components/RosTable/redux/systemsTableReducer.js:9]: I'm not a huge fan of switches as they tend to be too wordy and not so obvious, you can use `applyReducerHash` and have an object mapper

```JS
return applyReducerHash({
  'FETCH_CLOUD_SYSTEMS_LIST_PENDING': (state) => ({ ...state, loading: true }),
  'FETCH_CLOUD_SYSTEMS_LIST_FULFILLED': (state, { payload }) => ({ ...state, loading: false, systemsData: payload }),
  'FETCH_CLOUD_SYSTEMS_LIST_REJECTED': (state, { payload }) => ({ ...state, loading: false, systemError: payload }),
}, initialState)
```

File: src/Components/RosToolbar/RosToolbar.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Toolbar, ToolbarItem, ToolbarGroup, ToolbarContent, TextInput } from '@patternfly/react-core';
+import { TablePagination } from '../Pagination/Pagination';
+import SearchIcon from '@patternfly/react-icons/dist/js/icons/search-icon';
+//import propTypes from 'prop-types';
+
+export class RosToolbar extends React.Component {
```
Comment (karelhala) [inline src/Components/RosToolbar/RosToolbar.js:7]: No need to have class, function components should be prefered.

File: src/Components/RosToolbar/RosToolbar.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    constructor(props) {
+        super(props);
+    }
+
+    render() {
+        return (
+            <React.Fragment>
```
Comment (karelhala) [inline src/Components/RosToolbar/RosToolbar.js:14]: No need to wrap in `React.Fragment` as the component returns just `Toolbar` component with inner components.

**Comment by kgaikwad** [comment]
@karelhala, thank you for review. I am working on review comments.
With the current implementation in main branch [here](https://github.com/RedHatInsights/ros-frontend/blob/main/src/Routes/RosPage/RosPage.js#L29). I haven't added `await` for API response so after refreshing many times it is not showing records on PAGE but from network panel I can see records in response. 
I think with redux changes, it will render records properly so separating out pagination work from this PR and keeping only redux work.

File: src/Components/RosTable/redux/actions.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    return {
+        type: 'FETCH_CLOUD_SYSTEMS_LIST',
+        payload: fetch(url).then(handleErrors)
+        .then(res =>  res.json())
+        .then(json => { return json.results; })
+        .catch(error => { return error; })
+    };
+}
+
+function handleErrors(response) {
```
[the remark this answers is not in the document]
Comment (kgaikwad) [reply src/Components/RosTable/redux/actions.js:14]: Currently everyone is facing issue on table rendering so will it be good to push remaining changes first?
We will track axios library changes in a separate issue [RHIROS-56](https://issues.redhat.com/browse/RHIROS-56).
Any thoughts on this?

**Comment by apuntamb** [comment]
Tested locally and I do not observe any wait time to see the records on UI. Below is the screenshot attached for reference with browser console that shows no errors. Works like a charm! :+1: 

redux_pr

File: src/Components/RosTable/redux/systemsTableReducer.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+const initialState = {
+    loading: false,
+    systemsData: [],
+    systemError: {}
+};
+
+const systemsTableReducer = applyReducerHash({
+    FETCH_CLOUD_SYSTEMS_LIST_PENDING: (state) => ({ ...state, loading: true }),
+    FETCH_CLOUD_SYSTEMS_LIST_FULFILLED: (state, action) => {
+        console.log(action);
```
Comment (karelhala) [inline src/Components/RosTable/redux/systemsTableReducer.js:12]: Forgotten `console.log`

**Review by karelhala — APPROVED** [review]
Apart from one `console.log` looking good!

**Review by apuntamb — APPROVED** [review]

**Comment by apuntamb** [comment]
Merged, thanks @kgaikwad and @karelhala!
